sdk/test/Services/S3/IntegrationTests/PutObjectTests.cs

```diff
@@ -940,6 +940,96 @@ public void PutObject_StreamChecksumEnabled()
+                {
+                    BucketName = bucketName,
+                    AccessControlPolicy = new S3AccessControlList
+                    {
+                        Grants = new List<S3Grant>()
+                    {
+                        new S3Grant()
+                        {
+                            Grantee = new S3Grantee()
+                            {
+                                URI = "http://acs.amazonaws.com/groups/s3/LogDelivery",
+                            },
+                            Permission = S3Permission.READ
+                        },
+                    },
+                        Owner = getBucketAclResponse.Owner,
+                    },
+                });
+                var getBucketAclResponse2 = Client.GetBucketAcl(new GetBucketAclRequest
+                {
+                    BucketName = bucketName,
+                });
+                getBucketAclResponse2.Grants.ForEach(grant =>
+                {
+                    Assert.IsTrue(grant.Grantee.URI == "http://acs.amazonaws.com/groups/s3/LogDelivery");
+                    Assert.IsTrue(grant.Permission == S3Permission.READ);
+                });
+
+                Client.PutObject(new PutObjectRequest
+                {
+                    Key = "putobjectwithacl",
+                    ContentBody = "randomstuff",
+                    BucketName = bucketName,
+                });
+                Client.PutObjectAcl(new PutObjectAclRequest
+                {
+                    BucketName = bucketName,
+                    Key = "putobjectwithacl",
+                    AccessControlPolicy = new S3AccessControlList
+                    {
+                        Grants = new List<S3Grant>()
+                    {
+                        new S3Grant()
+                        {
+                            Grantee = new S3Grantee()
+                            {
+                                URI = "http://acs.amazonaws.com/groups/global/AuthenticatedUsers",
+                            },
+                            Permission = S3Permission.READ
+                        },
+                    },
+                        Owner = getBucketAclResponse.Owner,
+                    },
+                });
+                var getObjectAclResponse = Client.GetObjectAcl(new GetObjectAclRequest
+                {
+                    Key = "putobjectwithacl",
+                    BucketName = bucketName,
+                });
+
+                getObjectAclResponse.Grants.ForEach(grant =>
+                {
+                    Assert.IsTrue(grant.Grantee.URI == "http://acs.amazonaws.com/groups/global/AuthenticatedUsers");
+                    Assert.IsTrue(grant.Permission == S3Permission.READ);
+                });
+            }
+            finally 
+            {                 
+                AmazonS3Util.DeleteS3BucketWithObjects(Client, bucketName);
```

Comment:
I don't think this finally is correct, the `bucketName` variable here doesn't refer to the bucket you created but the `private static` variable with the same name: https://github.com/aws/aws-sdk-net/blob/e83c9f26e7fc2a45f3ce10a940ba64b8ffb7ff1d/sdk/test/Services/S3/IntegrationTests/PutObjectTests.cs#L33

I'd either use a different variable name in this method or update it not to create a new bucket at all (you'd need to confirm changing the ACLs doesn't impact the other tests here - as they all share the same bucket).

Reply:
thank you for pointing that out. I moved the variable outside the `try` block and renamed it. I also just ran the integ test and confirmed it is deleting the right bucket.